code/API_definitions/qos-profiles.yaml

```
@@ -47,33 +47,24 @@ tags:
 
 paths:
   /qos-profiles:
-    get:
+    post:
       tags:
         - QoS Profiles
-      summary: "Get All QoS Profiles"
+      summary: Get QoS profiles avaialable for a given device
       description: |
-        Returns all QoS Profiles that match the given criteria, or all profiles if no criteria is specified.
-
-        The access token may be either a 2-legged or 3-legged access token. If the access token is 3-legged, all returned QoS Profiles must be available to all end users associated with the access token.
-
-      operationId: getQosProfiles
-      security:
-        - openId:
-            - qos-profiles:qos-profiles:read
+        Returns all QoS Profiles that match the given criteria.
+        If the client credentials specify a device, the QoS Profiles returned are those that are applicable to the device.
+        If no criteria is given and the credentials don't specify a device, all QoS Profiles are returned.
```

"If the client credentials don't specify a device" is for me not clear enough, especially compared to the language we have in quality-on-demand and also previously here. If a 3-legged token is used the API client must only get profiles which are actually available to "all end users associated with the access token".

```suggestion
        Returns all QoS Profiles that match the given criteria.
        The access token may be either a 2-legged or 3-legged access token. If the access token is 3-legged, all returned QoS Profiles will be available to all end users associated with the access token. It
```

There is the need to add more documentation within the API description, cf #313